### PR TITLE
kitty: update to 0.11.1

### PR DESCRIPTION
**Pull request by emacsomancer**

### CODE REVIEW

**Comment by SolitudeSF** [comment]
sphinx should be in hostmakedeps

**Comment by emacsomancer** [comment]
@SolitudeSF  ah! thanks for that.

File: srcpkgs/kitty/template [hunk 1]
```diff
@@ -13,7 +13,7 @@ maintainer="Julio Galvan <juliogalvan@protonmail.com>"
 license="GPL-3.0-or-later"
 homepage="https://github.com/kovidgoyal/kitty"
```
Comment (SolitudeSF) [inline srcpkgs/kitty/template]: new homepage is `https://sw.kovidgoyal.net/kitty/`

File: srcpkgs/kitty/template [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # Template file for 'kitty'
 pkgname=kitty
-version=0.10.1
+version=0.11.1
 revision=1
 python_version=3
-hostmakedepends="ncurses pkg-config python3-devel wayland-devel wayland-protocols"
+hostmakedepends="ncurses pkg-config python3-devel wayland-devel wayland-protocols python3-Sphinx"
```
Comment (SolitudeSF) [inline srcpkgs/kitty/template]: `python3-devel wayland-devel wayland-protocols` dont need to be always present in the hostmakedepends, only for crossbuilding, you could add them conditionally like this
```
if [ -n "$CROSS_BUILD" ]; then
	hostmakedepends+=" python3-devel wayland-devel wayland-protocols"
fi
```

**Comment by SolitudeSF** [comment]
also squash commits in one and force push

**Comment by emacsomancer** [comment]
ok, everything should be changed, with squashed commits

**Comment by SolitudeSF** [comment]
```
if [ -n "$CROSS_BUILD" ]; then
	hostmakedepends+=" python3-devel wayland-devel wayland-protocols"
fi
```
should be outside the `do_build` not merged with it

**Comment by emacsomancer** [comment]
I still don't seem to be able to get the arm versions to build.

File: srcpkgs/kitty/template [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
 maintainer="Julio Galvan <juliogalvan@protonmail.com>"
 license="GPL-3.0-or-later"
-homepage="https://github.com/kovidgoyal/kitty"
+homepage="https://sw.kovidgoyal.net/kitty/"
 distfiles="https://github.com/kovidgoyal/kitty/archive/v${version}.tar.gz"
-checksum=ef22208497a76e2f88ebe56c176e4608f049b056252cf1bf122c9c1ec711cfa6
+checksum=3bbc6b5465d424969b16c5ad7f2f67ffbfe33657fdcb443e1bcc11aa00726841
+
+if [ -n "$CROSS_BUILD" ]; then
+	hostmakedepends+=" python3-devel wayland-devel wayland-protocols libxkbcommon xkeyboard-config xkeyboard-config libxkbfile libxkbcommon-devel fontconfig-devel kitty"
```
Comment (newbluemoon) [inline srcpkgs/kitty/template]: Please don’t exceed a line length of 80 characters for better readability. :)

File: srcpkgs/kitty/template [hunk 4]
```diff
@@ -1,24 +1,34 @@
...
 license="GPL-3.0-or-later"
-homepage="https://github.com/kovidgoyal/kitty"
+homepage="https://sw.kovidgoyal.net/kitty/"
 distfiles="https://github.com/kovidgoyal/kitty/archive/v${version}.tar.gz"
-checksum=ef22208497a76e2f88ebe56c176e4608f049b056252cf1bf122c9c1ec711cfa6
+checksum=3bbc6b5465d424969b16c5ad7f2f67ffbfe33657fdcb443e1bcc11aa00726841
+
+if [ -n "$CROSS_BUILD" ]; then
+	hostmakedepends+=" python3-devel wayland-devel wayland-protocols \
+	 libxkbcommon xkeyboard-config xkeyboard-config libxkbfile \
```
Comment (newbluemoon) [inline srcpkgs/kitty/template]: The backslash isn’t necessary here. :)

Comment (SolitudeSF) [reply srcpkgs/kitty/template]: and all these deps.

**Comment by SolitudeSF** [comment]
there were no changes to dependencies since 0.10.1 so i think the issue with crosscompilation is on the voids side.

File: srcpkgs/kitty/template (hunk 4 above)
Comment (emacsomancer) [reply srcpkgs/kitty/template]: because of the ""s ?

File: srcpkgs/kitty/template (hunk 3 above)
Comment (emacsomancer) [reply srcpkgs/kitty/template]: My editor spoils me for such things.....

**Comment by emacsomancer** [comment]
@SolitudeSF Yeah, I had expected this to be an easy version bump.

File: srcpkgs/kitty/template (hunk 4 above)
Comment (emacsomancer) [reply srcpkgs/kitty/template]: ( @SolitudeSF I had added some additional hostmakedepends to try to address the crosscompiliation errors )

**Comment by SolitudeSF** [comment]
yes, its definitely void's issue since i cant crosscompile the version from current void tree (`0.10.1`)

**Comment by emacsomancer** [comment]
Unfortunately, I'm getting the same sorts of build errors for the ARM builds.

File: srcpkgs/kitty/template [hunk 5]
```diff
@@ -1,24 +1,33 @@
...
+	hostmakedepends+=" python3-devel wayland-devel wayland-protocols
+	 libxkbcommon xkeyboard-config xkeyboard-config libxkbfile
+	 libxkbcommon-devel fontconfig-devel kitty"
+fi
 
 do_build() {
 	if [ -n "$CROSS_BUILD" ]; then
 		export CFLAGS="-I${XBPS_CROSS_BASE}/usr/include/python3.6m"
 		sed -i "s|-fpie|-fpie ${CFLAGS}|g" setup.py
+		sed -i -e 's;\(from kitty.constants import \
```
Comment (newbluemoon) [inline srcpkgs/kitty/template]: It doesn’t work because of this line break; it changes the pattern to be found and sed doesn’t do anything.

**Comment by emacsomancer** [comment]
looks like it's built successfully! the '2nd method' seems to work to produce documentation (& config files) properly, so the config file hack seems unnecessary. Thanks @newbluemoon  & @SolitudeSF for your help.

File: srcpkgs/kitty/template [hunk 6]
```diff
@@ -1,24 +1,32 @@
...
 maintainer="Julio Galvan <juliogalvan@protonmail.com>"
 license="GPL-3.0-or-later"
-homepage="https://github.com/kovidgoyal/kitty"
+homepage="https://sw.kovidgoyal.net/kitty/"
 distfiles="https://github.com/kovidgoyal/kitty/archive/v${version}.tar.gz"
-checksum=ef22208497a76e2f88ebe56c176e4608f049b056252cf1bf122c9c1ec711cfa6
+checksum=3bbc6b5465d424969b16c5ad7f2f67ffbfe33657fdcb443e1bcc11aa00726841
+
+if [ -n "$CROSS_BUILD" ]; then
+	hostmakedepends+=" python3-devel wayland-devel wayland-protocols
```
Comment (SolitudeSF) [inline srcpkgs/kitty/template]: you only need to add kitty here.

Comment (emacsomancer) [reply srcpkgs/kitty/template]: I'll try. After I got everything to build, I did experiment with removing some of the `hostmakedepends` without much luck. I'll try again though.

Comment (SolitudeSF) [reply srcpkgs/kitty/template]: i just tested it myself. it should work

Comment (SolitudeSF) [reply srcpkgs/kitty/template]: wait, no, thats not what i meant, sorry. im mistaking current template and my local. 
you need `hostmakedepends+=" python3-devel wayland-devel wayland-protocols kitty"`

Comment (SolitudeSF) [reply srcpkgs/kitty/template]: sorry, this is taking far more travis attempts because i cant word my comments properly.

**Comment by SolitudeSF** [comment]
travis says that only linting failed

**Comment by emacsomancer** [comment]
changed to source with prebuild docs. Additional hostmakedepends are required to get aarch64 to build.

File: srcpkgs/kitty/template [hunk 7]
```diff
@@ -1,19 +1,25 @@
...
-homepage="https://github.com/kovidgoyal/kitty"
-distfiles="https://github.com/kovidgoyal/kitty/archive/v${version}.tar.gz"
-checksum=ef22208497a76e2f88ebe56c176e4608f049b056252cf1bf122c9c1ec711cfa6
+homepage="https://sw.kovidgoyal.net/kitty/"
+distfiles="https://github.com/kovidgoyal/kitty/releases/download/v${version}/kitty-${version}.tar.xz"
+checksum=3baf72ee4de8818bbc205fd631f716b60eabc8bd215ec7e834e00a01db7912c4
+
+if [ -n "$CROSS_BUILD" ]; then
+	hostmakedepends+=" python3-devel wayland-devel wayland-protocols
+	 libxkbcommon xkeyboard-config xkeyboard-config libxkbfile
```
Comment (cr6git) [inline srcpkgs/kitty/template]: `xkeyboard-config xkeyboard-config`

twice the same?

Comment (SolitudeSF) [reply srcpkgs/kitty/template]: this only needs ` python3-devel wayland-devel wayland-protocols kitty`. xkeyboard-config issue is on voids side and other stuff is absolutely unnecessary.

Comment (SolitudeSF) [reply srcpkgs/kitty/template]: #477 is supposed to fix this

Comment (emacsomancer) [reply srcpkgs/kitty/template]: ok, rebuilding with only ` python3-devel wayland-devel wayland-protocols`

**Comment by emacsomancer** [comment]
You can see that without the extra `hostmakedepends` I had added (modulo removing the doubled ` xkeyboard-config` pointed out by @cr6git), the aarch64 versions don't build properly.

**Comment by SolitudeSF** [comment]
`xkeyboard-config` issue is on voids side and other stuff is absolutely unnecessary.
#477 is supposed to fix this

**Comment by emacsomancer** [comment]
So shall we just wait until [#477](https://github.com/void-linux/void-packages/pull/477) is pulled?

**Comment by SolitudeSF** [comment]
yes

**Comment by Johnnynator** [comment]
It compiles fine for `aarch64-musl` `index: added `kitty-0.11.1_1' (aarch64-musl).`
Imo this PR is fine as it is now.

**Pull request merged**